# HG changeset patch
# User alta88@fixall.com
# Date 1745143433 -7200
# Parent  351c94b3da19faef8b6bd1ce63fc1296b7cb8b6f
NNN25 - No unnecessary tooltips in tabs.

Authored by Alta88 and contributed to the Betterbird Project.
See https://github.com/Betterbird/thunderbird-patches/blob/main/LICENSE for license details.
All uses require attribution to the Author.

diff --git a/mail/base/content/widgets/tabmail-tab.js b/mail/base/content/widgets/tabmail-tab.js
--- a/mail/base/content/widgets/tabmail-tab.js
+++ b/mail/base/content/widgets/tabmail-tab.js
@@ -83,8 +83,35 @@
         },
         true
       );
 
+      // Prevent tooltip if unnecessary, ie. title is not overflowing.
+      this.addEventListener("mouseenter", event => {
+        const titleNode = event.target?.querySelector(".tab-content");
+        if (!titleNode?.title) {
+          return;
+        }
+
+        const textNode = titleNode.querySelector("label.tab-text");
+        if (titleNode.title != textNode?.textContent) {
+          return;
+        }
+
+        const textContainerNode = titleNode.querySelector(
+          ".tab-label-container"
+        );
+        if (textContainerNode.hasAttribute("textoverflow")) {
+          return;
+        }
+
+        titleNode.title = "";
+        this.addEventListener(
+          "mouseleave",
+          () => (titleNode.title = textNode.textContent),
+          { once: true }
+        );
+      });
+
       const closeButton = this.querySelector(".tab-close-button");
 
       // Prevent switching to the tab before closing it by stopping the
       // mousedown event.
